Makefile

```diff
@@ -1,13 +1,24 @@
-package: 
+.PHONY: clippy dev format lint package run scratch test ts_test test_application test_rust test_rehash
```

Comment:
Make is very much centered around the filesystem. The default assumption is that each target is a file or directory that will be created or updated when make runs. For example the assumption is that `make package` will create a file called `package`. To get the most out of make's dependency system you want to adhere to that assumption when you can.

It's also perfectly fine to have targets that do not correspond to files like you have here. (Those are called [phony targets](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html).) In that case the convention is to declare those targets with `.PHONY`. The reason is makes change detection: if you ever end up with a file that happens to have the same name as one of your phony targets, when you run the target make will do nothing if it thinks the file (or directory) hasn't changed. But if you declare the targets to be phony they will run regardless.